shaker/libs/pygit2_utils.py

```diff
@@ -0,0 +1,96 @@
+import shaker.libs.logger
+import pygit2
+import sys
+
+
+def pygit2_parse_error(e):
+    """
+    Parse a pygit2 specific error into a more understandable context. Will
+    also run some checks to try and help with the problem.
+
+    Args:
+        e(Exception): The exception that was raised
+    """
+    # Common errors to look for are,
+    # AttributeError: 'module' object has no attribute 'KeypairFromAgent'
+    # _pygit2.GitError: Unsupported URL protocol
+
+    # Set the logging level to noisy so we can generate as much usefult info as possible
+    shaker.libs.logger.Logger().basicConfig(level=shaker.libs.logger.Logger().INFO)
+    error_msg = ("shaker.libs.util:check_pygit2: Pygit2 problem detected, '%s'"
+                 % (e.message))
+    shaker.libs.logger.Logger().critical(error_msg)
+    pygit2_info()
+    shaker.libs.logger.Logger().critical("shaker.libs.util:check_pygit2: Running system checks for common issues")
+    pygit2_check()
+    sys.exit(1)
+
+
+def pygit2_info():
+    """
+    Output key pygit2/libgit2 information
+    """
+    link_versions = "http://www.pygit2.org/install.html#version-numbers"
+    message_versions = ("shaker.libs.util:check_pygit2: pygit2 *requires* the correct "
+                        "version of libgit2, this version was built against libgit2 version '%s'. "
+                        "Please check the versions on your system if you experience "
+                        "problems. (For compatibility, please refer to %s)"
+                        % (pygit2.LIBGIT2_VERSION, link_versions))
+    shaker.libs.logger.Logger().warning(message_versions)
+
+
+def pygit2_check():
+    """
+    Run all checks for pygit2 sanity
+    """
+    check = pygit2_check_ssh() and pygit2_check_credentials()
+    if check:
+        # Info that we've not found any known problems
+        message_ok = ("shaker.libs.util:check_pygit2: No common pygit2 problems found. ")
+        shaker.libs.logger.Logger().debug(message_ok)
```

Comment:
Given that each of the check functions already produce debug logs this extra log line here doesn't tell us anything extra so we should remove it